spec/rubocop/cop/rspec/repeated_example_group_description_spec.rb

```diff
@@ -0,0 +1,232 @@
+    expect_no_offenses(<<-RUBY)
+      RSpec.describe 'Animal', 'dog' do
+        # example group
+      end
+
+      RSpec.describe 'Animal', 'cat' do
+        # example group
+      end
+    RUBY
+  end
+
+  it 'does not register offense for describe block with several docstring' do
+    expect_no_offenses(<<-RUBY)
+      RSpec.describe 'Animal', 'dog', type: :request do
+        # example group
+      end
+
+      RSpec.describe 'Animal', 'cat', type: :request do
+        # example group
+      end
+    RUBY
+  end
+
+  it 'register offense for different example group with similar descriptions' do
+    expect_offense(<<-RUBY)
+      describe 'Animal' do
+      ^^^^^^^^^^^^^^^^^^^^ Repeated describe block description on line(s) [5]
+        # example group
+      end
+
+      context 'Animal' do
+      ^^^^^^^^^^^^^^^^^^^ Repeated context block description on line(s) [1]
+        # example group
+      end
+    RUBY
+  end
+
+  it 'registers offense only for RSPEC namespace example groups' do
+    expect_offense(<<-RUBY)
+      helpers.describe 'doing x' do
+        it { cool_predicate_method }
+      end
+
+      RSpec.describe 'doing x' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated describe block description on line(s) [9]
+        it { cool_predicate_method }
+      end
+
+      context 'doing x' do
+      ^^^^^^^^^^^^^^^^^^^^ Repeated context block description on line(s) [5]
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'registers offense only for RSPEC namespace example groups in any order' do
+    expect_offense(<<-RUBY)
+      RSpec.describe 'doing x' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated describe block description on line(s) [5]
+        it { cool_predicate_method }
+      end
+
+      context 'doing x' do
+      ^^^^^^^^^^^^^^^^^^^^ Repeated context block description on line(s) [1]
+        it { cool_predicate_method }
+      end
+
+      helpers.describe 'doing x' do
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'registers offense only for example group' do
+    expect_offense(<<-RUBY)
+      RSpec.describe 'A' do
+        stub_all_http_calls()
+        allowed_statuses = %i[open submitted approved].freeze
+        before { create(:admin) }
+
+        describe '#load' do
+        ^^^^^^^^^^^^^^^^^^^ Repeated describe block description on line(s) [10]
+          it { cool_predicate_method }
+        end
+
+        describe '#load' do
+        ^^^^^^^^^^^^^^^^^^^ Repeated describe block description on line(s) [6]
+          it { cool_predicate_method }
+        end
+      end
+    RUBY
+  end
+
+  it 'considers interpolated docstrings as different descriptions' do
+    expect_no_offenses(<<-RUBY)
+      context "when class is \#{A::B}" do
+        # ...
+      end
+
+      context "when class is \#{C::D}" do
+        # ...
+      end
+    RUBY
+  end
+
+  it 'registers offense correctly for interpolated docstrings' do
```

Comment:
Should it? if there is a variable and the variable value changes between the two context calls?
Although this is such an edge case that I could live with the false positives. Actually it would be very bad code to start with

Reply:
I hope there's no one in the world writing `describe "when time is #{Time.now.to_f}" do` and repeat that exact description :D It's better to flag this even though technically those two will be two different descriptions.